### ISSUE TITLE
Generation problem with jwt security

### ISSUE THREAD

**Issue by CyrilChev** [deepu105/generator-jhipster-bootswatch#7]
Hi,

I've used JHipster v3.3.0 to generate a monolithic project with JWT security. Then I used JHipster Bootswatch generator to add bootswatch support.

I've got a first problem when I start the client side:

`angular.js:4576 Uncaught Error: [$injector:unpr] Unknown provider: localStorageServiceProvider <- localStorageService <- BootswatchInterceptor <- $http <- $translatePartialLoader
`
The problem is in BootswatchInterceptor, so I changed the declaration of the interceptor from this :

`.factory('BootswatchInterceptor', function ($rootScope, $q, $location, localStorageService) {`

to this : 

`.factory('BootswatchInterceptor', ['$rootScope', '$q', '$location', '$localStorage', '$sessionStorage', function ($rootScope, $q, $location, $localStorage, $sessionStorage) {
`

The problem was solved, but then I got another error with an invalid http status code 405 for bootswatch request :

`XMLHttpRequest cannot load http://bootswatch.com/api/3.json?cacheBuster=1464258215905. Response for preflight has invalid HTTP status code 405`

When I looked to the code inside BootswatchInterceptor, the condition for testing bootswatch url does nothing. So I changed the code like this :

```
                if(config.url.indexOf('bootswatch.com/api') !== -1){
                    delete config.headers.Authorization;                   
                }

```

And then everything works perfectly.

I'm not an angularjs or a jhipster guru so I'm very prudent with this, but I think the generator should be adapted to reflect these changes.

Thank you.

**Comment by deepu105** [comment]
Your solution is good. Can you do a PR to fix this?

Thanks & Regards,
Deepu

**Comment by CyrilChev** [comment]
I've done a PR to submit changes.

Thanks for your answer.

Cyril

**Comment by deepu105** [comment]
closed via #8
